StudyBuddy/Controllers/ProfileController.cs

```diff
@@ -112,8 +118,87 @@ public async Task<IActionResult> SaveProfile(string name, string birthdate, stri
         }
         catch (Exception ex)
         {
-            ErrorViewModel errorModel = new ErrorViewModel { ErrorMessage = "Error uploading file: " + ex.Message, };
+            ErrorViewModel errorModel = new() { ErrorMessage = "Error uploading file: " + ex.Message };
             return View("Error", errorModel);
         }
     }
+
+    public IActionResult Login(string? userId)
+    {
+        UserId? currentUserId = _userService.GetCurrentUserId();
+
+        if (currentUserId != null && _userManager.GetUserById(currentUserId.Value) != null)
+        {
+            return RedirectToAction("Index", "Home");
+        }
+
+        if (string.IsNullOrEmpty(userId))
+        {
```

Comment:
`TempData["ErrorMessage"] = "User ID not provided";` maybe?

Reply:
This would cause the error to pop up every time you enter the page as the textbox is empty.